services/web/server/src/simcore_service_webserver/payments/_methods_db.py

```diff
@@ -0,0 +1,239 @@
+from pydantic import BaseModel, PositiveInt, parse_obj_as
+from simcore_postgres_database.models.payments_methods import (
+    InitPromptAckFlowState,
+    payments_methods,
+)
+from sqlalchemy import literal_column
+from sqlalchemy.sql import func
+
+from ..db.plugin import get_database_engine
+from .errors import (
+    PaymentMethodAlreadyAckedError,
+    PaymentMethodNotFoundError,
+    PaymentMethodUniqueViolationError,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+#
+class PaymentsMethodsDB(BaseModel):
+    payment_method_id: PaymentMethodID
+    user_id: UserID
+    wallet_id: WalletID
+    # State in Flow
+    initiated_at: datetime.datetime
+    completed_at: datetime.datetime | None
+    state: InitPromptAckFlowState
+    state_message: str | None
+
+    class Config:
+        orm_mode = True
+
+
+async def insert_init_payment_method(
+    app: web.Application,
+    *,
+    payment_method_id: str,
+    user_id: UserID,
+    wallet_id: WalletID,
+    initiated_at: datetime.datetime,
+) -> None:
+    async with get_database_engine(app).acquire() as conn:
+        try:
+            await conn.execute(
+                payments_methods.insert().values(
+                    payment_method_id=payment_method_id,
+                    user_id=user_id,
+                    wallet_id=wallet_id,
+                    initiated_at=initiated_at,
+                )
+            )
+        except db_errors.UniqueViolation as err:
+            raise PaymentMethodUniqueViolationError(
+                payment_method_id=payment_method_id
+            ) from err
+
+
+async def list_payments_methods(
+    app,
+    *,
+    user_id: UserID,
+    wallet_id: WalletID,
+    offset: PositiveInt | None = None,
+    limit: PositiveInt | None = None,
+) -> tuple[int, list[PaymentsMethodsDB]]:
+    """List payments entries done by a give user (in any state)
+
+    Sorted by newest-first
+    """
+
+    async with get_database_engine(app).acquire() as conn:
+        total_number_of_items = await conn.scalar(
+            sa.select(sa.func.count())
+            .select_from(payments_methods)
+            .where(
+                (payments_methods.c.user_id == user_id)
+                & (payments_methods.c.wallet_id == wallet_id)
+            )
+        )
+        assert total_number_of_items is not None  # nosec
+
+        query = (
+            payments_methods.select()
+            .where(
+                (payments_methods.c.user_id == user_id)
+                & (payments_methods.c.wallet_id == wallet_id)
+            )
+            .order_by(payments_methods.c.created.desc())
+        )  # newest first
+
+        # NOTE: what if between these two calls there are new rows? can we get this in an atomic call?
+        if offset is not None:
+            if offset > total_number_of_items:
+                msg = f"{offset=} exceeds {total_number_of_items=}"
+                raise ValueError(msg)
+
+            query.offset(offset)
```

Comment:
isn't this necessary?
```suggestion
            query = query.offset(offset)
```

Reply:
aha good catch. 
No, this is not necessary. I will remove the entire function.
It was a trial i did but we do not use it.  thx for noticing